controllers/tf_controller.go

```diff
@@ -256,6 +247,22 @@ func (r *TerraformReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		conditions.MarkUnknown(terraform, meta.ReadyCondition, meta.ProgressingReason, "Reconciliation in progress")
 	}
 
+	// Check whether we need to reconcile the release at this time
+	if shouldReconcile, reason, requeueAfter := r.shouldReconcile(terraform, sourceObj); !shouldReconcile {
+		log.Info("Skipping reconciliation. Interval has not elapsed since last plan",
+			"lastPlanAt", terraform.Status.LastPlanAt,
+			"nextReconcile", terraform.Status.LastPlanAt.Add(terraform.Spec.Interval.Duration),
+			"requeueAfter", requeueAfter,
+			"reason", reason)
```

Comment:
@alexandermarston  The function only sets non-empty reason when it returns `true`, but this log is only printed when function returned `false`, so it seems that the reason will never actually be used?